src/interface/container_collection.hpp

```diff
@@ -57,10 +82,16 @@ class ContainerCollection {
       c.second->Print();
       std::cout << std::endl;
     }
+    for (auto &sc : swarmContainers_) {
+      std::cout << "SwarmContainer " << sc.first << " has:" << std::endl;
+      sc.second->Print();
+      std::cout << std::endl;
+    }
   }
 
  private:
   std::map<std::string, std::shared_ptr<Container<T>>> containers_;
+  std::map<std::string, std::shared_ptr<SwarmContainer>> swarmContainers_;
```

Comment:
Why are you using an map and not an unordered_map here? Do you need to preserve the ordering? unordered_map will be more efficient.

Reply:
I just copied `map` from its usage in `containers_`; if we want to change these I would change both but I don't want to mess with that in this PR. My guess now though is that there isn't much cost associated with these data structures.